src/sidebar/services/groups.js

```diff
@@ -225,75 +234,68 @@ function groups(
-            // If there is a direct-linked group, add it to the featured groups list.
-            const allFeaturedGroups =
-              selectedGroup !== null &&
-              !featuredGroups.some(g => g.id === selectedGroup.id)
-                ? featuredGroups.concat([selectedGroup])
-                : featuredGroups;
-
-            // If there's a selected annotation it may require an extra api call
-            // to fetch its group.
-            if (selectedAnn) {
-              // Set the directLinkedAnnotationGroupId to be used later in
-              // the filterGroups method.
-              directLinkedAnnotationGroupId = selectedAnn.group;
-
-              const selectedAnnGroup = myGroups
-                .concat(allFeaturedGroups)
-                .some(g => g.id === selectedAnn.group);
-
-              // If the direct-linked annotation's group has not already been fetched,
-              // fetch it.
-              if (!selectedAnnGroup) {
-                return fetchGroup({
-                  id: selectedAnn.group,
-                  expand: params.expand,
-                }).then(directLinkedAnnGroup => {
-                  // If the directLinkedAnnotation's group fetch failed, return
-                  // the list of groups without it.
-                  if (!directLinkedAnnGroup) {
-                    return [
-                      combineGroups(myGroups, allFeaturedGroups, documentUri),
-                      token,
-                    ];
-                  }
-
-                  // If the directLinkedAnnotation's group fetch was successful,
-                  // combine it with the other groups.
-                  return [
-                    combineGroups(
-                      myGroups,
-                      allFeaturedGroups.concat(directLinkedAnnGroup),
-                      documentUri
-                    ),
-                    token,
-                  ];
-                });
+        return Promise.all(groupApiRequests);
+      })
+      .then(([myGroups, featuredGroups, token, selectedAnn, selectedGroup]) => {
+        // Step 3. Add the direct-linked group to the list of featured groups,
+        // and if there was a direct-linked annotation, fetch its group if we
+        // don't already have it.
+
+        // If there is a direct-linked group, add it to the featured groups list.
+        let allFeaturedGroups =
+          selectedGroup !== null &&
+          !featuredGroups.some(g => g.id === selectedGroup.id)
+            ? featuredGroups.concat([selectedGroup])
+            : featuredGroups;
+
+        // If there's a selected annotation it may require an extra api call
+        // to fetch its group.
+        if (selectedAnn) {
+          // Set the directLinkedAnnotationGroupId to be used later in
+          // the filterGroups method.
+          directLinkedAnnotationGroupId = selectedAnn.group;
+
+          const selectedAnnGroup = myGroups
```

Comment:
I'm having trouble with the "selected" vs. "directLinked" terminology again here...

`selectedAnnGroup` is a Boolean, right (return value of `Array.prototype.some`)? The naming here is confusing as it implies an object/thing.

I feel like this line could use a comment along the lines of "see if the selected annotation's group is already in our known list of groups" (assuming I'm even parsing that correctly :) )

Reply:
I agree. I'm not sure why I didn't see this as an issue when I reviewed the original code. I've pushed a change to use the term "direct-linked" consistently.